File: app/src/main/java/com/airfreshener/telegram_sms/model/ReplyMarkupKeyboard.java
package com.airfreshener.telegram_sms.model;

import java.util.ArrayList;

public class ReplyMarkupKeyboard {
    public static ArrayList<InlineKeyboardButton> getInlineKeyboardObj(String text, String callbackData) {
        InlineKeyboardButton button = new InlineKeyboardButton();
        button.text = text;
        button.callback_data = callbackData;
        ArrayList<InlineKeyboardButton> buttonArraylist = new ArrayList<>();
        buttonArraylist.add(button);
        return buttonArraylist;
    }

    @SuppressWarnings("unused")
    public static class KeyboardMarkup {
        public ArrayList<ArrayList<InlineKeyboardButton>> inline_keyboard;
        boolean one_time_keyboard = true;
    }

    public static class InlineKeyboardButton {
        String text;
        String callback_data;
    }
}

File: app/src/main/java/com/airfreshener/telegram_sms/model/ReplyMarkupKeyboard.kt
package com.airfreshener.telegram_sms.model

object ReplyMarkupKeyboard {
    fun getInlineKeyboardObj(
        vararg data: Pair<String?, String?>
    ): List<List<InlineKeyboardButton>> = data.map { (text, callbackData) ->
        InlineKeyboardButton().apply {
            this.text = text
            this.callback_data = callbackData
        }.let { listOf(it) }
    }

    @Suppress("PropertyName", "unused")
    class KeyboardMarkup {
        @JvmField
        var inline_keyboard: List<List<InlineKeyboardButton>>? = null
        @JvmField
        var one_time_keyboard = true
    }

    @Suppress("PropertyName")
    class InlineKeyboardButton {
        var text: String? = null
        var callback_data: String? = null
    }
}
